cdap-app-fabric/src/main/java/io/cdap/cdap/internal/namespace/credential/handler/GcpWorkloadIdentityHttpHandler.java

```diff
@@ -0,0 +1,294 @@
+import io.cdap.cdap.internal.credential.CredentialIdentityManager;
+import io.cdap.cdap.internal.credential.CredentialProfileManager;
+import io.cdap.cdap.internal.namespace.credential.GcpWorkloadIdentityUtil;
+import io.cdap.cdap.proto.NamespaceMeta;
+import io.cdap.cdap.proto.credential.CredentialIdentity;
+import io.cdap.cdap.proto.credential.CredentialProfile;
+import io.cdap.cdap.proto.credential.CredentialProvider;
+import io.cdap.cdap.proto.credential.IdentityValidationException;
+import io.cdap.cdap.proto.credential.NamespaceWorkloadIdentity;
+import io.cdap.cdap.proto.id.CredentialIdentityId;
+import io.cdap.cdap.proto.id.CredentialProfileId;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.security.spi.authentication.SecurityRequestContext;
+import io.cdap.cdap.security.spi.authorization.ContextAccessEnforcer;
+import io.cdap.cdap.security.spi.authorization.UnauthorizedException;
+import io.cdap.http.AbstractHttpHandler;
+import io.cdap.http.HttpHandler;
+import io.cdap.http.HttpResponder;
+import io.netty.buffer.ByteBufInputStream;
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.HttpRequest;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.Optional;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+/**
+ * {@link HttpHandler} for namespace identity providers.
+ */
+@Singleton
+@Path(Gateway.API_VERSION_3)
+public class GcpWorkloadIdentityHttpHandler extends AbstractHttpHandler {
+  private static final Gson GSON = new Gson();
+
+  private static final String SYSTEM_PROFILE_NAME = "ns-gcp-wi";
+  private static final String CREDENTIAL_PROVIDER_NAME = "gcp-wi-credential-provider";
+
+  private final ContextAccessEnforcer accessEnforcer;
+  private final NamespaceQueryAdmin namespaceQueryAdmin;
+  private final CredentialIdentityManager credentialIdentityManager;
+  private final CredentialProfileManager credentialProfileManager;
+  private final CredentialProvider credentialProvider;
+
+  @Inject
+  GcpWorkloadIdentityHttpHandler(ContextAccessEnforcer accessEnforcer,
+      NamespaceQueryAdmin namespaceQueryAdmin,
+      CredentialIdentityManager credentialIdentityManager,
+      CredentialProfileManager credentialProfileManager,
+      CredentialProvider credentialProvider) throws BadRequestException, IOException {
+    this.accessEnforcer = accessEnforcer;
+    this.namespaceQueryAdmin = namespaceQueryAdmin;
+    this.credentialIdentityManager = credentialIdentityManager;
+    this.credentialProfileManager = credentialProfileManager;
+    this.credentialProvider = credentialProvider;
+
+    // create the system profile if not exists.
+    createSystemProfileIfNotExists();
```

Comment:
Generally we don't want to create a resource in the constructor, it can lead to all kinds of problems. Can we instead model this as a capability? Or add it to the initialization step of a service.